### PR TITLE
feat(MarkdownPreview): Add Html support in mkd editor

### PR DESCRIPTION
**Pull request by Hugo-Galley**
This Pr follow issues #413 

### Add:

- Accept HTML in the editor
- Sanitize the __*DOM*__ to prevent __*XSS*__ vulnerabilities

### CODE REVIEW

File: ui/leafwiki-ui/src/features/preview/MarkdownPreview.tsx [hunk 1]
```diff
@@ -129,10 +131,10 @@ export default function MarkdownPreview({ content, path }: Props) {
     <>
       <ReactMarkdown
         remarkPlugins={[remarkGfm, remarkLineNumber]}
-        rehypePlugins={[rehypeHighlight]}
+        rehypePlugins={[rehypeRaw, rehypeHighlight]}
         components={components}
       >
-        {content}
+        {DOMPurify.sanitize(content)}
```
[the remark this answers is not in the document]
Comment (Hugo-Galley) [reply ui/leafwiki-ui/src/features/preview/MarkdownPreview.tsx]: I'm not okay with this response. The content is sanitized and it works.
It helps make the code easier to understand, and for the user, it allows them to write the tags they need without being able to inject scripts.

**Review by perber — CHANGES_REQUESTED** [review]
Hey @Hugo-Galley,

thanks for your PR!

I noticed an issue in the sanitized output.
Your approach works for most cases, but there’s a corner case where code blocks containing HTML. 

Example: 
```
```html
<script>alert("test")</script>
```

get stripped.
Since DOMPurify runs before the Markdown parsing, it removes the `<script>` tags and the code block content doesn’t show up.

**I’ve attached a screenshot showing how I tested the behavior:**

Image

File: ui/leafwiki-ui/package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "@types/codemirror": "^5.60.15",
     "class-variance-authority": "^0.7.1",
     "clsx": "^2.1.1",
+    "dompurify": "^3.3.1",
```
Comment (perber) [inline ui/leafwiki-ui/package.json]: Can we remove this dependency?

**Comment by perber** [comment]
> @perber Indeed, I hadn't thought about that case. So I changed the approach, and the content is now sanitized after the Markdown rendering.
> 
> ### This allows us to get the desired result:
> image


Looks good now! Great job! One minor thing: Can we get rid of the dependency dompurify?

File: ui/leafwiki-ui/package.json (hunk 2 above)
Comment (Hugo-Galley) [reply ui/leafwiki-ui/package.json]: Yes of course, I removed this now

**Review by perber — APPROVED** [review]
lgtm!
I have to fix the print view in the dark mode. 
When I have fixed it. I will roll it out and create a new release.
